src/pages/ClaimsPage/ClaimsPage.tsx

```diff
@@ -80,9 +86,18 @@ export default function ClaimsPage() {
       </SingleColumnLayout>
     );
 
-  const claimsGroupByVault = groupBy(
-    claims.sort(c => c.id),
-    c => c.distribution.epoch.circle?.id && c.distribution.vault.vault_address
+  const claimedClaimsGroupByRow = groupBy(
```

Comment:
could clean this up into one function handling both cases

Reply:
i think you could also just have one data structure, now that you're also grouping by txHash?

```
c1 = claims from hasura
c2 = claimRows(c1)
c3 = groupBy(c1.sort(c => -c.id), claimsRowKey) 
```

then when you loop through either `c2.filter(c => c.txHash)` or `c2.filter(c => !c.txHash)`, its corresponding group is already in `c3`

it's now apparent that c2 is actually completely redundant; instead of using that, you could do e.g. `Object.values(c3).filter(group => !group[0].txHash)` to get the unclaimed groups, then sort by the min id of each group to get the rows in order

this is probably a bit bikesheddy; no need to do it this way if what you have now works; but i hope it was at least informative